genesyscloud/integration_function_action/genesyscloud_integration_function_action_init_test.go

```diff
@@ -0,0 +1,66 @@
+package integration_function_action
+
+import (
+	"sync"
+	"testing"
+
+	integration "terraform-provider-genesyscloud/genesyscloud/integration"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+/*
+   The genesyscloud_integration_function_action_init_test.go file is used to initialize the data sources and resources
+   used in testing the integration action resource.
+
+   Please make sure you register ALL resources and data sources your test cases will use.
+*/
+
+// providerDataSources holds a map of all registered datasources
+var providerDataSources map[string]*schema.Resource
+
+// providerResources holds a map of all registered resources
+var providerResources map[string]*schema.Resource
+
+type registerTestInstance struct {
+	resourceMapMutex   sync.RWMutex
+	datasourceMapMutex sync.RWMutex
+}
+
+// registerTestResources registers all resources used in the tests
+func (r *registerTestInstance) registerTestResources() {
+	r.resourceMapMutex.Lock()
+	defer r.resourceMapMutex.Unlock()
+
+	providerResources["genesyscloud_integration_function_action"] = ResourceIntegrationFunctionAction()
+	providerResources["genesyscloud_integration"] = integration.ResourceIntegration()
```

Comment:
Use `providerResource[integration.ResourceType] = ...` to ensure the resource type name is consistent